framework/doc/content/source/kernels/ConservativeAdvection.md

```diff
@@ -22,27 +22,97 @@ element solution of the weak formulation. The first term is the volumetric term
+following way.  For each $i$, the quantity
+\begin{equation}
+\tilde{R}_{i} = -(\nabla\psi_i, \vec{v}) \ ,
+\end{equation}
+is evaluated.  If $\tilde{R}_{i}>0$ is positive then mass (or heat, or whatever $u$
+represents) is flowing *out* of node $i$, and this is called an
+"upwind" node.  If $\tilde{R}_{i}\geq 0$, the residual for node $i$
+is set to
+\begin{equation}
+R_{i} = u_{i}\tilde{R}_{i} \ \ \texttt{ for } \ \ \tilde{R}_{i}\geq 0.
+\end{equation}
+Here $u_{i}$ is the value of $u$ at the node $i$.  Define the total
+mass flowing from the upwind nodes:
+\begin{equation}
+M_{\mathrm{out}} = \sum_{i\ \mathrm{ with }\ \tilde{R}_{i}\geq 0}
+u_{i}\tilde{R}_{i} \ .
+\end{equation}
+Similarly, define
+\begin{equation}
+\tilde{M}_{\mathrm{in}} = - \sum_{i \ \mathrm{ with }\  \tilde{R}_{i}< 0}
+\tilde{R}_{i} \ .
+\end{equation}
+Mass is conserved if the residual for the downwind nodes is defined to
+be
+\begin{equation}
+R_{i} = \tilde{R}_{i} M_{\mathrm{out}} / M_{\mathrm{in}}  \ \ \texttt{ for } \ \ \tilde{R}_{i}< 0.
+\end{equation}
+
+## Dirichlet boundary conditions for pure advection
+
+In the purely advective case, Dirichlet boundary conditions may be
+used to inject mass (or heat, or whatever $u$ corresponds to
+physically) into the domain, or extract it from the domain.
+Specifically:
+
+- if $\vec{v}\cdot\vec{n}<0$ (with $\vec{n}$ being the outward normal) then the velocity field is "blowing" mass into the domain.  Fixing $u=u_{B}$ on this boundary means the mass flux (units kg.s$^{-1}$.m$^{-2}$) entering the domain is $-\vec{v}\cdot\vec{n}u_{B}$.
+- if $\vec{v}\cdot\vec{n}>0$ (with $\vec{n}$ being the outward normal) then the velocity field is "blowing" mass out of the domain.  If no Dirichlet boundary condition is used then mass is prevented from leaving the domain via this boundary: mass arriving here will "build up", causing $u$ to increase.  Alternatively, if $u=0$ is fixed at the boundary all mass arriving here will exit the domain immediately.
+
+
 ## Example Syntax
 
 `ConservativeAdvection` can be used in a variety of problems, including
 advection-diffusion-reaction. The syntax for `ConservativeAdvection` is
-demonstrated in this `Kernels` block from an advection-diffusion-reaction test
-case:
+demonstrated in this `Kernels` block from a pure advection test case:
 
-!listing
-test/tests/dgkernels/advection_diffusion_mixed_bcs_test_resid_jac/dg_advection_diffusion_test.i
-block=Kernels label=false
+!listing test/tests/kernels/conservative_advection/no_upwinding_1D.i block=Kernels
 
-The velocity is supplied as a three component vector with order $v_x$, $v_y$, and  $v_z$.
+The velocity is supplied as a three component vector with order $v_x$,
+$v_y$, and  $v_z$.
+
+## Comparison of no upwinding and full upwinding
+
+The tests
+!listing test/tests/kernels/conservative_advection/no_upwinding_1D.i
+and
+!listing test/tests/kernels/conservative_advection/full_upwinding_1D.i
```

Comment:
I am curious: this problem will not reach stead state solution u=1?

Reply:
I see. lim(t->infty) u(x) = 1 on all x inside the domain except at x on the right boundary. Is du/dx at the right boundary equal to zero?

Reply:
In the limit t->infty, u(x)=1 for x<1, but u(1)=infty.

Reply:
Please see the extended comment above, @YaqiWang